pandas/core/arrays/_utils.py

```diff
@@ -44,7 +44,16 @@ def to_numpy_dtype_inference(
         dtype_given = True
 
     if na_value is lib.no_default:
-        na_value = arr.dtype.na_value
+        if dtype is None:
```

Comment:
Could you use `na_value_for_dtype`?

Reply:
Maybe 'm wrong, but `na_value_for_dtype` might be a bit too general in that it returns `np.nan` for everything else and changes behavior:

from:

```
In [4]: pd.Series([True, pd.NA], dtype="boolean").to_numpy()
Out[4]: array([True, <NA>], dtype=object)
```
to:
```
In [4]: pd.Series([True, pd.NA], dtype="boolean").to_numpy()
Out[4]: array([True, nan], dtype=object)
```

Reply:
Ah OK that's fair. Thanks for checking